x-pack/plugins/observability_ai_assistant/public/hooks/use_genai_connectors.ts

```diff
@@ -32,11 +33,9 @@ export function useGenAIConnectors(): UseGenAIConnectorsResult {
 
   const [error, setError] = useState<Error | undefined>(undefined);
 
-  useEffect(() => {
-    setLoading(true);
-
-    const controller = new AbortController();
+  const controller = useMemo(() => new AbortController(), []);
```

Comment:
The useEffect at line 71 needs to have access to `controller` as part of cleanup.

The change pulls the contents of the useEffect into a separate function so it can be reused for the new `reloadConnectors` method.